### PR TITLE
Reset active matrix marker in `\draw_transform_matrix_reset:`

### PR DESCRIPTION
**Pull request by muzimuzhi**
Need a Changelog entry?

### CODE REVIEW

File: l3experimental/l3draw/l3draw-transforms.dtx [hunk 1]
```diff
@@ -110,10 +110,11 @@
 %    \begin{macrocode}
 \cs_new_protected:Npn \draw_transform_matrix_reset:
   {
-    \fp_set:Nn \l_@@_matrix_a_fp { 1 }
+    \fp_set_eq:NN \l_@@_matrix_a_fp \c_one_fp
     \fp_zero:N \l_@@_matrix_b_fp
     \fp_zero:N \l_@@_matrix_c_fp
-    \fp_set:Nn \l_@@_matrix_d_fp { 1 }
+    \fp_set_eq:NN \l_@@_matrix_d_fp \c_one_fp
```
Comment (josephwright) [inline l3experimental/l3draw/l3draw-transforms.dtx:116]: I see the thinking here but I'm not so keen. In general, the balance between `\<var>_set:Nn` and `\<var>_set_eq:NN` for numerical data is tricky - I tend to think `set` for places you have an explicit value, `set_eq` when you are doing save-and-restore or similar: makes it clear what's going on. Here, I get why you want to use `set_eq` as there is a definite performance gain. But `l3draw` isn't exactly fast and this is a relatively 'high level' function. So I think the current approach has clarity on it's side: but I am happy to hear counter-arguments.

File: l3experimental/l3draw/l3draw-transforms.dtx [hunk 2]
```diff
@@ -110,10 +110,11 @@
...
+    \bool_set_false:N \l_@@_matrix_active_bool
```
Comment (josephwright) [inline l3experimental/l3draw/l3draw-transforms.dtx:117]: This part is nice and clear: definitely needed :)

**Review by josephwright — APPROVED** [review]
I'll see what others think before merging: as I say, my major comment is a matter of taste

**Comment by FrankMittelbach** [comment]
> I'll see what others think before merging: as I say, my major comment is a matter of taste

I'm usually in favor of speed improvements, but here I guess I'm with Joseph in both cases

**Comment by josephwright** [comment]
@FrankMittelbach favours speed: I'll merge

**Comment by FrankMittelbach** [comment]
looks like I wasn't clear ... I "usually favor" but I agree with both your comments so I wouldn't do the first part

**Comment by FrankMittelbach** [comment]
but I guess either are fine so ...

**Comment by josephwright** [comment]
@FrankMittelbach  OK, I've backed out and sse can see if there are more comments :)

File: l3experimental/l3draw/l3draw-transforms.dtx (hunk 1 above)
Comment (muzimuzhi) [reply l3experimental/l3draw/l3draw-transforms.dtx:116]: If these initial values are named constants like `\c__draw_matrix_scale_init_fp` and `\c__draw_matrix_slant_init_fp`, using `\fp_set_eq:NN` is more obvious.

Comment (josephwright) [reply l3experimental/l3draw/l3draw-transforms.dtx:116]: It's mainly used inside `l3fp` for return values, then also for comparisons

Comment (Skillmon) [reply l3experimental/l3draw/l3draw-transforms.dtx:116]: Then why document it? Do you see my point? I don't know how to further communicate it... I agree that I'd not use `\fp_set_eq:NN` for arbitrary initial values, but in the case of a constant following (like `\c_one_fp`) things are arguably different. It is instantly apparent, that if I set an `fp` equal to a constant I don't want to temporarily store the value of another variable to later restore it, but that I want to assign an initial value for something else!

**Pull request merged**